### PR TITLE
Implement the `serve` operator and `/serve` endpoint

### PR DESCRIPTION
**Pull request by dominiklohmann**
The `serve` sink operator lifts the results of a pipeline into an endpoint that can be fetched from incrementally. Its corresponding `/serve` endpoint is the replacement of the existing `/query` endpoint.

```[tasklist]
### Tasks
- [x] Implement the `serve` sink operator
- [x] Implement the `/serve` endpoint
- [x] Implement the `serve-manager` component
- [x] Connect the three parts
- [x] Support configuring the buffer size as part of the `serve` operator
- [x] Document the data flow between the three parts
- [x] Remove the `/query` endpoint
- [x] Document the operator and write a changelog entry
- [x] Write tests
```

### CODE REVIEW

File: libvast/builtins/operators/serve.cpp [hunk 1]
```diff
@@ -0,0 +1,941 @@
...
+//
+// The web is a lossy place—which is why the serve-manager caches its last
+// result set and the last continuation token. To also be able to cache the last
+// result set, we delay the removal of the managed serve operators in the
+// serve-manager by 1 minute.
+//
+// Technically, the serve-manager should not be needed. However, the current
+// architecture of the web plugin makes it so that the REST handler actor is not
+// implicitly a component actor, and as such may run outside of the node or even
+// multiple times. We should revisit this in the future.
```
Comment (dominiklohmann) [inline libvast/builtins/operators/serve.cpp:46]: I recommend reading this at the very start of the review.

File: plugins/web/integration/scripts/serve.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+jq -nec "${first_result} | .events | length == 1"
+
+# Query the second set.
+second_result="$(curl -XPOST -H "Content-Type: application/json" -d "{\"serve_id\": \"version\", \"continuation_token\": \"${continuation_token}\", \"timeout\": \"5s\", \"max_events\": 4}" http://127.0.0.1:42025/api/v0/serve)"
+
+# The next set should be the reamining 90 results.
+jq -nec "${second_result} | .next_continuation_token == null"
+jq -nec "${second_result} | .events | length == 4"
+
+wait
```
Comment (dominiklohmann) [inline plugins/web/integration/scripts/serve.sh:23]: This is a bit of a hack to get tests into our current integration testing framework; we could add more this way, but it's not really well maintainable.

**Review by Dakostu — APPROVED** [review]
I played around with this and it seems to work, so: approval from me.

I did leave some comments though.

File: plugins/web/integration/scripts/serve.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+first_result="$(curl -XPOST -H "Content-Type: application/json" -d '{"serve_id": "version", "timeout": "5s", "max_events": 1}' http://127.0.0.1:42025/api/v0/serve)"
+continuation_token="$(jq -rnec "${first_result} | .next_continuation_token")"
+
+# The first set should be 10 results exactly.
+jq -nec "${first_result} | .events | length == 1"
+
+# Query the second set.
+second_result="$(curl -XPOST -H "Content-Type: application/json" -d "{\"serve_id\": \"version\", \"continuation_token\": \"${continuation_token}\", \"timeout\": \"5s\", \"max_events\": 4}" http://127.0.0.1:42025/api/v0/serve)"
+
+# The next set should be the reamining 90 results.
```
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply plugins/web/integration/scripts/serve.sh:19]: Fixed by 39cfdd439c

File: libvast/builtins/operators/serve.cpp [hunk 4]
```diff
@@ -0,0 +1,941 @@
...
+                                         "{}; got parameters {}",
+                                         serve_id.error(), rq.params));
+    }
+    if (not *serve_id) {
+      return caf::make_error(ec::invalid_argument,
+                             fmt::format("serve_id must be specified; got "
+                                         "parameters {}",
+                                         rq.params));
+    }
+    result.serve_id = std::move(**serve_id);
```
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply libvast/builtins/operators/serve.cpp:587]: Fixed by 7c6d4ff5a5

File: plugins/web/src/server_command.cpp
[the remark this answers is not in the document]
Comment (dominiklohmann) [reply plugins/web/src/server_command.cpp:1]: Fixed by 0e7de19f3f

**Pull request merged**